src/HasEnums.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Spatie\Enum;
+
+use Spatie\Enum\Exceptions\InvalidEnumError;
+use Spatie\Enum\Exceptions\InvalidValueException;
+
+trait HasEnums
+{
+    /**
+     * @param $key
+     * @param \Spatie\Enum\Enum $enumObject
+     *
+     * @return mixed
+     */
+    public function setAttribute($key, $enumObject)
+    {
+        if (! isset($this->enums[$key])) {
+            return parent::setAttribute($key, $enumObject);
+        }
+
+        $enumClass = $this->enums[$key];
+
+        if (! is_a($enumObject, $enumClass)) {
```

Comment:
Why don't use the `Enum::make()` method here to allow `$model->status = 'archived';`